File: design-patterns/java/State-DesignPattern/src/WithStatePattern/CashDispensedState.java
package WithStatePattern;

public class CashDispensedState implements ATMState {

    public void insertCard(ATMMachine atm) {
        System.out.println("Transaction in progress");
    }

    public void enterPin(ATMMachine atm, int pin) {
        System.out.println("Transaction in progress");
    }

    public void withdrawCash(ATMMachine atm, int amount) {
        System.out.println("Transaction completed");
    }

    public void ejectCard(ATMMachine atm) {
        System.out.println("Please take your card");
        atm.setState(new IdleState());
    }
}

File: design-patterns/python/State-DesignPattern/src/with_state_pattern/cash_dispensed_state.py
"""CashDispensed state - cash given, waiting for card to be taken."""
from .atm_state import ATMState
from .atm_machine import ATMMachine


class CashDispensedState(ATMState):
    def insert_card(self, atm: ATMMachine) -> None:
        print("Transaction in progress")

    def enter_pin(self, atm: ATMMachine, pin: int) -> None:
        print("Transaction in progress")

    def withdraw_cash(self, atm: ATMMachine, amount: int) -> None:
        print("Transaction completed")

    def eject_card(self, atm: ATMMachine) -> None:
        from .idle_state import IdleState
        print("Please take your card")
        atm.set_state(IdleState())
